sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/transfer/BlockDownloadEntity.java

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.storage.blob.transfer;
+
+import androidx.room.ColumnInfo;
+import androidx.room.Entity;
+import androidx.room.ForeignKey;
+import androidx.room.Ignore;
+import androidx.room.Index;
+import androidx.room.PrimaryKey;
+import androidx.room.TypeConverters;
+
+import com.azure.android.core.util.Base64Util;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+
+import static androidx.room.ForeignKey.CASCADE;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/**
+ * Package private.
+ *
+ * Represents metadata for a block download.
+ *
+ * There is one-to-many relationship between a {@link BlobDownloadEntity} and a set of {@link BlockDownloadEntity}.
+ * Each {@link BlockDownloadEntity} in the set describes a chunk of the file referenced by the
+ * {@link BlobDownloadEntity}
+ *
+ * The Data Access Object type {@link DownloadDao} exposes DB store and read methods on this model.
+ *
+ * @see TransferDatabase
+ */
+@Entity(tableName = "blockdownloads",
+    foreignKeys = @ForeignKey(entity = BlobDownloadEntity.class,
+        parentColumns = "key",
+        childColumns = "blob_key",
+        onDelete = CASCADE),
+    indices = {@Index("blob_key")})
+final class BlockDownloadEntity {
+    /**
+     * A unique key for the block download metadata.
+     *
+     * This key identifies the metadata in the local store, which is different from
+     * {@link BlockDownloadEntity#blockId}. Block ID is used by the storage service to uniquely identify the block.
+     */
+    @PrimaryKey(autoGenerate = true)
+    @ColumnInfo(name = "key")
+    public Long key;
+
+    /**
+     * The key of {@link BlobDownloadEntity} that this {@link BlockDownloadEntity} belongs to.
+     */
+    @ColumnInfo(name = "blob_key")
+    public long blobKey;
+
+    /**
+     * The absolute path to the file that the block is a part of.
+     */
+    @ColumnInfo(name = "file_path")
+    public String filePath;
+
+    /**
+     * The offset in the file from which block contents starts.
+     */
+    @ColumnInfo(name = "blob_offset")
+    public long blobOffset;
+
+    /**
+     * The block size in bytes.
+     */
+    @ColumnInfo(name = "block_size")
+    public long blockSize;
+
+    /**
+     * The base64 block ID used by the storage service to uniquely identify the block.
+     */
+    @ColumnInfo(name = "block_id")
+    public String blockId;
```

Comment:
we don't send it to service in download case, this will be used only for tracking running calls